### PR TITLE
Retrieve results working for zip

### PR DESCRIPTION
**Pull request by segerphilip**
Zip file creation happens but does not complete before the files are sent by the server, which obviously does not work. Working on a fix for that. Please do not close/merge yet.

### CODE REVIEW

**Comment by segerphilip** [comment]
Fixed, now working for zips as expected

File: app/handlers/retrieveResults.py
Comment (dcelik) [inline app/handlers/retrieveResults.py]: why is this identifier[0]? this should be just identifier. If this is wrong, it is due to a bug in the GUI cloud_api code.

File: app/handlers/retrieveResults.py
Comment (dcelik) [inline app/handlers/retrieveResults.py]: Lets define this chunk size somewhere up above for easier changing?

File: app/handlers/retrieveResults.py
Comment (dcelik) [inline app/handlers/retrieveResults.py]: Should retrieve results have a post function? If so, it should do something like returning a helpful error message

File: app/handlers/retrieveResults.py
Comment (dcelik) [inline app/handlers/retrieveResults.py]: Lets implement the basehandler exception handling here.

File: app/handlers/retrieveResults.py
Comment (dcelik) [inline app/handlers/retrieveResults.py]: Will need to inherit from baseHandler to do error handling the way we want.

File: app/handlers/retrieveResults.py
Comment (dcelik) [inline app/handlers/retrieveResults.py]: Do we want to keep the zip file after its been created? It might be prudent to delete it after its done. Take a look at uploadVideo to see how to do the on_finish for removing temp files. The only benefit I can see is that we can return it immediately if it has already been zipped, but this code does not implement such a check. This might be worth discussing with @runnersaw , @jabb1123 , @youralien .

**Review by dcelik — CHANGES_REQUESTED** [review]
Looks mostly good. I have a few small fixes and some higher level questions that should be addressed by somebody on the team.

File: app/handlers/retrieveResults.py
Comment (segerphilip) [reply app/handlers/retrieveResults.py]: This was a bug, not sure what the cause was.

**Comment by dcelik** [comment]
Fixed merge conflicts due to other PR being merged. Should be retested @segerphilip before approval.

File: app/handlers/retrieveResults.py
Comment (segerphilip) [reply app/handlers/retrieveResults.py]: Does it often need to be changed? But yeah, probably.

File: app/handlers/retrieveResults.py
Comment (jabb1123) [reply app/handlers/retrieveResults.py]: Is there a way to do some form of checksum testing on the individual files to know if the content has changed. If so, then re-zip and send those changed files. Otherwise, send nothing.

File: app/handlers/retrieveResults.py
Comment (dcelik) [reply app/handlers/retrieveResults.py]: This may be something that we want to experiment with later in order to find a good chunk size

File: app/handlers/retrieveResults.py
Comment (dcelik) [reply app/handlers/retrieveResults.py]: Damn, didn't think of that. Maybe we create a checksum that gets held and do a status check using the statusHandler framework?

Comment (segerphilip) [reply app/handlers/retrieveResults.py]: Holding off on this for now

File: app/handlers/retrieveResults.py
Comment (runnersaw) [inline app/handlers/retrieveResults.py]: y tho

File: app/handlers/retrieveResults.py
Comment (runnersaw) [inline app/handlers/retrieveResults.py]: probably use `os.path.basename` here to get the name for the file rather than `file_path[dir_len:]`

File: app/handlers/retrieveResults.py
Comment (runnersaw) [reply app/handlers/retrieveResults.py]: https://i.ytimg.com/vi/Wr2iBbApeq0/maxresdefault.jpg

File: app/handlers/retrieveResults.py
Comment (runnersaw) [inline app/handlers/retrieveResults.py]: I'm semi worried about infinite loops here (reason being I tried to do a similar thing where I'd break when there were no images in a directory, and that broke). Do you have any ideas about stopping conditions other than `if not data`?

**Review by runnersaw — COMMENTED** [review]
Mostly good to go, just a few style things mostly.

File: app/handlers/retrieveResults.py
Comment (dcelik) [reply app/handlers/retrieveResults.py]: 👍

File: app/handlers/retrieveResults.py
Comment (jabb1123) [reply app/handlers/retrieveResults.py]: @runnersaw You said that you had a similar bit of code. Could you possible share your solution to this problem? If not applicable, ignore this.

Comment (runnersaw) [reply app/handlers/retrieveResults.py]: I just learned empty strings are `False`-y, and `f.read()` returns an empty string if at the end of file. This works for me!

**Review by runnersaw — APPROVED** [review]
🚢 it!

**Review by dcelik — CHANGES_REQUESTED** [review]
Lets remove the zip file after we send it. See uploadVideo.on_finish()

**Review by dcelik — APPROVED** [review]
Looks good, ship it!

**Pull request merged**
